src/analysis/AnalysisClient.js

```diff
@@ -208,6 +220,32 @@ class AnalysisClient {
     */
     return profitablePairedProducts;
   }
+
+
+  /*
+    Does profitability analysis of item based on lowest price offered
+    Input_params: PairedProductsList object of potentially profitable items
+    Returns: PairedProductsList object of profitable items.
+  */
+  getSecondaryAnalysis(pairedProductsList) {
+    /*
+      for each product, compare its total cost with the lowest offer price
+      if its price gives us ROI >= threshold_ROI%, we can include it in the to-buy list
+      At this point, we are taking minimal risk. It is possible to still buy and sell
+      quickly if our ROI < 25% but we can look into that after we establish profitability
+      with this cautious approach     
+    */
+    let that = this;
+    let analyzedProduct;
+    let profitablePairedProducts = new PairedProductList();       
+    pairedProductsList.products.forEach(function(pairedProduct){
+      analyzedProduct = that._getAnalyzedProductInfo(pairedProduct);
```

Comment:
Don't store this in a middle man variable if you don't need to. Use it in the conditional directly.
```
if (that._getAnalyzedProductInfo(pairedProduct).basePercentROI >= that.ROIThreshold) {
```